control-plane/csi-driver/src/bin/node/fsfreeze/bin/ioctl.rs

```diff
@@ -0,0 +1,61 @@
+use crate::fsfreeze::bin::FsFreezeOpt;
+use nix::{errno::Errno, ioctl_readwrite};
+use std::{fs, os::fd::AsRawFd, path::PathBuf};
+use tracing::error;
+
+// Nix macro to issue FIFREEZE ioctl.
+ioctl_readwrite!(fs_freeze, 'X', 119, i32);
+// Nix macro to issue FITHAW ioctl.
+ioctl_readwrite!(fs_thaw, 'X', 120, i32);
+
+/// Preflight checks to ensure that fsfreeze call would be made in a desirable condition.
+pub fn fsfreeze_preflight_check(volume_id: &str, nqn: &str) -> Result<(), Errno> {
+    let subsystems =
+        nvmeadm::nvmf_subsystem::Subsystem::try_from_nqn(nqn).map_err(|_| Errno::ENODATA)?;
+    if subsystems.iter().any(|subsystem| subsystem.state == "live") {
+        return Ok(());
+    }
+    error!(
+        volume_id = volume_id,
+        nqn = nqn,
+        "Preflight check for fsfreeze failed, subsystem has no live path"
+    );
+    Err(Errno::EHOSTUNREACH)
+}
+
+/// Issue FIFREEZE and FITHAW ioctl on the mount_path based on the command.
+pub async fn fsfreeze_ioctl(command: FsFreezeOpt, mount_path: PathBuf) -> Result<(), Errno> {
+    let blocking_task = tokio::task::spawn_blocking(move || {
+        let file = fs::File::open(&mount_path).map_err(|error| {
+            error!(
+                "Failed to get file descriptor path: {}, {}",
+                mount_path.display(),
+                error.to_string()
```

Comment:
```suggestion
                error,
                mount = mount_path.display(),
                "Failed to open volume mount path"
```

Reply:
```
error!(
                error = error.to_string(),
                mount = mount_path.display().to_string(),
                "Failed to open volume mount path"
            );
```

We need to convert to string other wise doesn't work like the above.

Reply:
Just add the formatter,
```suggestion
error!(
                %error,
                mount = %mount_path.display(),
                "Failed to open volume mount path"
            );
```

Reply:
Ahh right, I forgot about this, thanks. :-;